compiler/catala_utils/pos.ml

```diff
@@ -96,15 +96,10 @@ let to_string (pos : t) : string =
 
 let to_string_short (pos : t) : string =
   let s, e = pos.code_pos in
-  if e.Lexing.pos_lnum = s.Lexing.pos_lnum then
-    Printf.sprintf "%s:%d.%d-%d:" s.Lexing.pos_fname s.Lexing.pos_lnum
-      (s.Lexing.pos_cnum - s.Lexing.pos_bol)
-      (e.Lexing.pos_cnum - e.Lexing.pos_bol)
-  else
-    Printf.sprintf "%s:%d.%d-%d.%d:" s.Lexing.pos_fname s.Lexing.pos_lnum
-      (s.Lexing.pos_cnum - s.Lexing.pos_bol)
-      e.Lexing.pos_lnum
-      (e.Lexing.pos_cnum - e.Lexing.pos_bol)
+  Printf.sprintf "%s:%d.%d-%d.%d:" s.Lexing.pos_fname s.Lexing.pos_lnum
+    (s.Lexing.pos_cnum - s.Lexing.pos_bol + 1)
+    e.Lexing.pos_lnum
+    (e.Lexing.pos_cnum - e.Lexing.pos_bol + 1)
```

Comment:
The same-line format was too tedious to parse ?
(this is a subset of the [GNU format](https://www.gnu.org/prep/standards/standards.html#Formatting-Error-Messages) so I was assuming it would be straight-forward everywhere ; might have been worth adding this pointer in a comment, on second thought)

Well-spotted for the off-by-one, indeed the standard is to start from col. 1 and I was starting from 0. :+1:

Reply:
I didn't know there was a GNU standard for error messages, we should follow it and it should definitely in a comment! The same-line format that was here before made it slightly more complex to parse error messages (because you had to check whether the position was on the same line or not), so I just made it more simple to parse.